src/service/viewer-impl.js

```diff
@@ -1012,8 +1012,13 @@ export class Viewer {
           eventType, data, awaitResponse));
     }
 
+    // If there is no viewer, no further action is needed.
     if (!this.messagingReadyPromise_) {
-      return Promise.reject(getChannelError());
+      if (awaitResponse) {
+        return Promise.reject(getChannelError());
+      } else {
+        return Promise.resolve();
```

Comment:
:-) this is what i did here: https://github.com/ampproject/amphtml/pull/6525/files

But after discussion, we think it does not make sense to queue up the message if there will ever be a messaging chanel